src/index.js

```diff
@@ -1,37 +1,117 @@
 import './style.css';
+import {
+  renderToDo, edit,
+} from './modules/functionality.js';
 
-const activities = [
-  {
-    description: 'Firts',
-    completed: false,
-    id: 1,
-  },
-  {
-    description: 'Second',
-    completed: false,
-    id: 2,
-  },
-  {
-    description: 'Thrid',
+const form = document.getElementById('list-form');
+const containerTask = document.getElementById('toDo-list');
+const newToDo = document.getElementById('imput-task');
+let toDos = [];
+
+// function add taks
+const addTask = () => {
+  const toDoValue = newToDo.value;
+  const emptyToDo = toDoValue === '';
+
+  if (emptyToDo) return;
+
+  const task = {
+    value: toDoValue,
     completed: false,
-    id: 3,
-  },
-];
-
-const containerTask = document.getElementById('task-list');
-
-const showTaskList = () => {
-  activities.forEach((task) => {
-    const card = document.createElement('div');
-    card.classList = 'task-description';
-    card.innerHTML = `
-    <div class="task-text">
-    <input type="checkbox">
-    <p class="task-text">${task.description}</p>
-    </div>
-    <i class="fa-solid fa-ellipsis-vertical"></i>`;
-    containerTask.appendChild(card);
-  });
+    id: toDos.length + 1,
+  };
+
+  newToDo.value = '';
+  toDos.push(task);
+  localStorage.setItem('todos', JSON.stringify(toDos));
+
+  renderToDo(); // llamamos a la función renderToDo para mostrar el nuevo elemento en la pantalla
 };
 
-showTaskList();
\ No newline at end of file
+// const addTask = () => {
+//   const toDoValue = newToDo.value;
+//   const emptyToDo = toDoValue === '';
+
+//   if (emptyToDo) return;
+//   const task = {
+//     value: toDoValue,
+//     completed: false,
+//     id: toDos.length + 1,
+//   };
+//   newToDo.value = '';
+//   toDos.push(task);
+//   localStorage.setItem('todos', JSON.stringify(toDos));
+// };
+const submitIcon = document.getElementById('submit-icon');
+submitIcon.addEventListener('click', addTask);
+
+// form submit
+form.addEventListener('submit', (event) => {
+  event.preventDefault();
+
+  addTask();
+  renderToDo();
+  localStorage.setItem('todos', JSON.stringify(toDos));
+});
+
+// delete task
+containerTask.addEventListener('click', (event) => {
+  if (event.target.classList.contains('delete-task-icon')) {
+    const taskElement = event.target.parentNode;
+    const taskId = parseInt(taskElement.querySelector('.text-input').getAttribute('data-id'), 10);
+    const taskIndex = toDos.findIndex((task) => task.id === taskId);
+    toDos.splice(taskIndex, 1);
+    taskElement.remove();
+
+    // update the id of the remaining tasks
+    // eslint-disable-next-line no-plusplus
+    for (let i = taskIndex; i < toDos.length; i++) {
```

Comment:
- [ ] Please make sure that you do not disable any eslint since eslint guides you to write clean code. I think you should remove this `   // eslint-disable-next-line no-plusplus` and change the `i++` to `i+=1` will fix your issue.